pkg/proxy/userspace/proxier.go

```diff
@@ -216,7 +237,21 @@ func createProxier(loadBalancer LoadBalancer, listenIP net.IP, iptables iptables
 		proxyPorts:      proxyPorts,
 		makeProxySocket: makeProxySocket,
 		exec:            exec,
-	}, nil
+	}
+	burstSyncs := 2
+	klog.V(3).Infof("minSyncPeriod: %v, syncPeriod: %v, burstSyncs: %d", minSyncPeriod, syncPeriod, burstSyncs)
+	proxier.syncRunner = async.NewBoundedFrequencyRunner("sync-runner", proxier.syncProxyRules, minSyncPeriod, syncPeriod, burstSyncs)
+
+	go func() {
+		<-loadBalancer.EndpointsSyncedChan()
```

Comment:
Why does this only wait for endpoints and not services?  If that's intentional, it could be commented.

Reply:
@thockin services are waited for via OnServicesSynced(), but because the userspace Proxier doesn't implement the EndpointsHandler interface (instead the RoundRobin LB does) we have to use a different way to figure out if endpoints are synced hence the passthrough.

The iptables proxier implements both ServicesHandler and EndpointsHandler so it doesn't have this issue.

Reply:
@thockin re-implemented this so that the proxy now implements EndpointsHandler and can intercept the OnEndpointsSynced().